src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionAzureKeyVaultProvider.java

```diff
@@ -64,9 +73,58 @@ public String getName() {
     }
 
     /**
-     * Constructor that authenticates to AAD. This is used by KeyVaultClient at runtime to authenticate to Azure Key
+     * Constructor that takes a callback function to authenticate to AAD. This is used by KeyVaultClient at runtime to authenticate to Azure Key
+     * Vault.
+     * 
+     * This constructor is present to maintain backwards compatibility with 6.0 version of the driver. Deprecated for removal in next Stable release.
+     * 
+     * @param authenticationCallback
+     *            - Callback function used for authenticating to AAD.
+     * @param executorService
+     *            - The ExecutorService, previously used to create the keyVaultClient, but not in use anymore. - This param can be passed as 'null'
+     * @throws SQLServerException
+     *             when an error occurs
+     */
+    @Deprecated
+    public SQLServerColumnEncryptionAzureKeyVaultProvider(SQLServerKeyVaultAuthenticationCallback authenticationCallback,
+            ExecutorService executorService) throws SQLServerException {
```

Comment:
Since we aren't doing anything with the executor, and the constructor is just here for backwards compatibility, shouldn't we make a call to `SQLServerColumnEncryptionAzureKeyVaultProvider(SQLServerKeyVaultAuthenticationCallback authenticationCallback)`? This might confuse people who are stepping through the code, thinking there is a very small change somewhere and that's why there are separate functions.

Reply:
Fixed